### PR TITLE
411 normalize function names between parts

### PR DESCRIPTION
**Pull request by ClaireHzl**
**Goto :** 
- Arm : combination of goto_joints and goto_from_matrix 
- Head : combination of goto_joints and goto_quat

**Get_current_state :** 
- new function name to replace get_joints_positions (arm and head), head.get_orientation (with the parameter as_quat=True), the hand.opening, the mobile_base.odometry

**Goto_posture :** 
Normalization with the add of the parameter "common_posture = "default"" for the head too ([issue 402](https://github.com/pollen-robotics/reachy2-sdk/issues/402)) 

**Modification of mobile_base.set_speed() :** 
Split in 2 functions : set_goal_speed(self, x: float | int = 0, y: float | int = 0, theta: float | int = 0) & send_speed_command()

### CODE REVIEW

File: src/reachy2_sdk/parts/head.py [hunk 1]
```diff
@@ -89,99 +89,92 @@ def __repr__(self) -> str:
...
-
-    def look_at(
-        self, x: float, y: float, z: float, duration: float = 2.0, wait: bool = False, interpolation_mode: str = "minimum_jerk"
-    ) -> GoToId:
-        """Compute and send neck rpy position to look at the (x, y, z) point in Reachy cartesian space (torso frame).
+        if as_quat:
+            quat = self._stub.GetOrientation(self._part_id).q
+            if round_int is not None:
+                return pyQuat(
+                    w=np.round(quat.w, round_int),
```
Comment (FabienDanieau) [inline src/reachy2_sdk/parts/head.py:103]: You don't want to round values inside a quaternion. Ever.

File: src/reachy2_sdk/parts/head.py [hunk 2]
```diff
@@ -89,99 +89,92 @@ def __repr__(self) -> str:
     def neck(self) -> Orbita3d:
         return self._neck
 
-    def get_orientation(self) -> pyQuat:
-        """Get the current orientation of the head.
+    def get_current_state(
+        self, degrees: bool = True, round_int: Optional[int] = None, as_quat: bool = False
```
Comment (FabienDanieau) [inline src/reachy2_sdk/parts/head.py:93]: not  big fan of `round_int`, at first I understood round as int. round isn't good enough?
also i'm not sure about the rounding in general. There are some in the __repr__ which are only there to make prints pretty.

**Review by FabienDanieau — CHANGES_REQUESTED** [review]
having a single goto funtion is nice!
There are some rounding with the quaternions that can't be done though

File: src/reachy2_sdk/parts/head.py (hunk 2 above)
Comment (ClaireHzl) [reply src/reachy2_sdk/parts/head.py:93]: I had issues with the name round, in files where the function np.round() was import as round(). That's why I changed it

**Comment by ClaireHzl** [comment]
Suppression of the "round", add of the overload goto functions and modification of the get_current_state() in different functions

**Review by FabienDanieau — APPROVED** [review]
Fantastic job!

**Pull request merged**